be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala

```diff
@@ -2,28 +2,82 @@ package ch.epfl.pop.pubsub.graph
 
 import akka.NotUsed
 import akka.stream.scaladsl.Flow
+import ch.epfl.pop.model.network.method.{Broadcast, Catchup, Publish, Subscribe, Unsubscribe}
+import ch.epfl.pop.model.network.{JsonRpcRequest, JsonRpcResponse}
 import ch.epfl.pop.model.network.requests.lao.{JsonRpcRequestCreateLao, JsonRpcRequestStateLao, JsonRpcRequestUpdateLao}
 import ch.epfl.pop.model.network.requests.meeting.{JsonRpcRequestCreateMeeting, JsonRpcRequestStateMeeting}
 import ch.epfl.pop.model.network.requests.witness.JsonRpcRequestWitnessMessage
-import ch.epfl.pop.pubsub.graph.validators.LaoValidator.{validateCreateLao, validateStateLao, validateUpdateLao}
-import ch.epfl.pop.pubsub.graph.validators.MeetingValidator.{validateCreateMeeting, validateStateMeeting}
-import ch.epfl.pop.pubsub.graph.validators.RollCallValidator.{validateCloseRollCall, validateCreateRollCall, validateOpenRollCall, validateReopenRollCall}
-import ch.epfl.pop.pubsub.graph.validators.WitnessValidator.validateWitnessMessage
+
+import ch.epfl.pop.pubsub.graph.validators.RpcValidator._
+import ch.epfl.pop.pubsub.graph.validators.MethodValidator._
+import ch.epfl.pop.pubsub.graph.validators.MessageValidator._
+import ch.epfl.pop.pubsub.graph.validators.LaoValidator._
+import ch.epfl.pop.pubsub.graph.validators.MeetingValidator._
+import ch.epfl.pop.pubsub.graph.validators.RollCallValidator._
+import ch.epfl.pop.pubsub.graph.validators.WitnessValidator._
 
 object Validator {
 
+  private val VALIDATOR_ERROR: PipelineError = PipelineError(
+    ErrorCodes.INVALID_ACTION.id,
+    "Internal server fault: Validator was given a message it could not recognize"
+  )
+
   // FIXME implement schema
   def validateSchema(jsonString: JsonString): Either[JsonString, PipelineError] = Left(jsonString)
 
-  // FIXME implement rpc validator
-  def validateJsonRpcContent(graphMessage: GraphMessage): GraphMessage = graphMessage match {
+  private def validateJsonRpcContent(graphMessage: GraphMessage): GraphMessage = graphMessage match {
     case Left(jsonRpcMessage) => jsonRpcMessage match {
-      case _ => ???
+      case message@(_: JsonRpcRequest) => validateRpcRequest(message)
+      case message@(_: JsonRpcResponse) => validateRpcResponse(message)
+      case _ => Right(VALIDATOR_ERROR)
+    }
+    case graphMessage@_ => graphMessage
+  }
+
+  private def validateMethodContent(graphMessage: GraphMessage): GraphMessage = graphMessage match {
+    case Left(jsonRpcRequest: JsonRpcRequest) => jsonRpcRequest.getParams match {
+      case _: Broadcast => validateBroadcast(jsonRpcRequest)
+      case _: Catchup => validateCatchup(jsonRpcRequest)
+      case _: Publish => validatePublish(jsonRpcRequest)
+      case _: Subscribe => validateSubscribe(jsonRpcRequest)
+      case _: Unsubscribe => validateUnsubscribe(jsonRpcRequest)
+      case _ => Right(VALIDATOR_ERROR)
+    }
+    case Left(_) => Right(PipelineError(
+      ErrorCodes.INVALID_ACTION.id,
```

Comment:
probably an internal error rather than an invalid action here?

Reply:
Truuuue because a response message ends up here, it's a bug in the server, not the client's fault!